File: project_euler2.js
// Solution to Project Euler Problem 2. Calculates the even-valued terms in 
// the Fibonacci sequence below a limit (default: 4000000)

function sumFibonacci(limit) {
    var sum = 0;
    var evenSum = 0;
    var term1 = 0; 
    var term2 = 0;
    for (var i = 1; i < limit; i = sum) {
        term1 = term2
        term2 = i;
        sum = term1 + term2;
        if (sum % 2 === 0) {
            evenSum += sum;
        }
    }
    return evenSum;
}

console.log(sumFibonacci(4000000));
        
        
        

File: project_euler2.rb
# Solution to Project Euler Problem 2. Calculates the even-valued terms in 
# the Fibonacci sequence below a limit (default: 4000000)

def sum_fibonacci(limit) 
    sum = 0
    even_sum = 0
    term1 = 0
    term2 = 0
    i = 1
    while i < limit
        term1 = term2
        term2 = i
        sum = term1 + term2
        i = sum
        if sum % 2 == 0
            even_sum += sum
        end
    end
    return even_sum
end

puts sum_fibonacci(4000000)
        
        
        
